tests/components/samsungtv/test_media_player.py

```diff
@@ -583,37 +509,23 @@ def mock_now():
     assert remote.control.call_args_list == [call("KEY_CHDOWN"), call("KEY")]
 
 
-async def test_turn_on_with_mac(hass, remote, wakeonlan):
-    """Test turn on."""
-    await setup_samsungtv(hass, MOCK_CONFIG)
-    assert await hass.services.async_call(
-        DOMAIN, SERVICE_TURN_ON, {ATTR_ENTITY_ID: ENTITY_ID}, True
-    )
-    # key and update called
-    assert wakeonlan.send_magic_packet.call_count == 1
-    assert wakeonlan.send_magic_packet.call_args_list == [
-        call("38:f9:d3:82:b4:f1", ip_address="255.255.255.255")
-    ]
-
-
-async def test_turn_on_with_mac_and_broadcast(hass, remote, wakeonlan):
+async def test_turn_on_with_turnon(hass, remote):
     """Test turn on."""
-    await setup_samsungtv(hass, MOCK_CONFIG_BROADCAST)
-    assert await hass.services.async_call(
-        DOMAIN, SERVICE_TURN_ON, {ATTR_ENTITY_ID: ENTITY_ID_BROADCAST}, True
-    )
-    # key and update called
-    assert wakeonlan.send_magic_packet.call_count == 1
-    assert wakeonlan.send_magic_packet.call_args_list == [
-        call("38:f9:d3:82:b4:f1", ip_address="192.168.5.255")
-    ]
+    with patch(
+        "homeassistant.components.samsungtv.media_player.Script.async_run"
+    ) as delay:
+        await setup_samsungtv(hass, MOCK_CONFIG)
+        assert await hass.services.async_call(
+            DOMAIN, SERVICE_TURN_ON, {ATTR_ENTITY_ID: ENTITY_ID}, True
+        )
+        assert delay.call_count == 1
```

Comment:
@MartinHjelmare How to test the async function. This is not working with unittest 3.7 - at least 3.8 is needed.

Reply:
I think it's this:
https://github.com/home-assistant/home-assistant/pull/28306#discussion_r363039139